txauthor.go

```diff
@@ -113,6 +136,46 @@ func (tx *TxAuthor) EstimateMaxSendAmount() (*Amount, error) {
 	}, nil
 }
 
+func (tx *TxAuthor) UseInputs(utxoKeys []string) error {
+	// first clear any previously set inputs
+	// so that an outdated set of inputs isn't used if an error occurs from this function
+	tx.inputs = nil
+	accountUtxos, err := tx.sourceWallet.UnspentOutputs(int32(tx.sourceAccountNumber))
+	if err != nil {
+		return fmt.Errorf("error reading unspent outputs in account: %v", err)
+	}
+
+	retrieveAccountUtxo := func(utxoKey string) *UnspentOutput {
+		for _, accountUtxo := range accountUtxos {
+			if accountUtxo.OutputKey == utxoKey {
+				return accountUtxo
+			}
+		}
+		return nil
+	}
```

Comment:
Instead of getting all utxos and looping through the slice each time you want to find a utxo, why don't you add a function to find utxo by transaction outpoint?.   
You can create output using the hash and index(you can get that by splitting `utxoKey` by ":") and get a single utxo the way it's done here: https://github.com/planetdecred/dcrlibwallet/blob/f3785c80f2f344a10f07715c3bac9ac4ede37712/accounts.go#L144-L147